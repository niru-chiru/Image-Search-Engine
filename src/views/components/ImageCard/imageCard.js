import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardMedia from "@material-ui/core/CardMedia";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";

const styles = (theme) => ({
  card: {
    minWidth: "100%",
    maxWidth: "100%",
    minHeight: "auto",
    borderRadius: 10,
    padding: 5,
  },
  content: {
    padding: 5,
  },
  media: {
    padding: 10,
    borderRadius: 10,
    paddingTop: 0,
    paddingLeft: 0,
    paddingRight: 0,
    paddingBottom: 0,
    marginBottom: 15,
  },
  title: {
    fontSize: 16,
    fontWeight: "bold",
    textAlign: "left",
  },
});

const ImageCard = (props) => {
  const { classes, imageurl, title } = props;

  return (
    <>
      <Card className={classes.card}>
        <CardContent className={classes.content}>
          <CardMedia
            component="img"
            alt=""
            className={classes.media}
            height="340"
            image={imageurl}
            title=""
          />
          <br />
          <Typography className={classes.title} color="textSecondary">
            {title}
          </Typography>
        </CardContent>
      </Card>
    </>
  );
};

ImageCard.propTypes = {
  classes: PropTypes.object.isRequired,
};
export default withStyles(styles)(ImageCard);
