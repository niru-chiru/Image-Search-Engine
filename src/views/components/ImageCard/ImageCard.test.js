import { render, screen } from "@testing-library/react";
import ImageCard from "./imageCard";

const title = "Sports!";
const imageurl = "";

it("matches snapshot", () => {
  const { asFragment } = render(
    <ImageCard title={title} imageurl={imageurl} />
  );
  expect(asFragment()).toMatchSnapshot();
});
