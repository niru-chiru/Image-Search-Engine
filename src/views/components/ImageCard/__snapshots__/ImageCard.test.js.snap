// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root ImageCard-card-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root ImageCard-content-2"
    >
      <img
        alt=""
        class="MuiCardMedia-root ImageCard-media-3 MuiCardMedia-media MuiCardMedia-img"
        height="340"
        title=""
      />
      <br />
      <p
        class="MuiTypography-root ImageCard-title-4 MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        Sports!
      </p>
    </div>
  </div>
</DocumentFragment>
`;
